File: project3/Team3/projectBQ/src/main/java/bitc/full502/projectbq/dto/UserDto.java
package bitc.full502.projectbq.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserDto {

    // 부서
    private long deptId;
    private String deptCode;
    private String deptName;

    // 사원
    private long empId;
    private String empCode;
    private String empName;
    private String empEmail;
    private String empPhone;
    private LocalDate empBirthDate;
    private LocalDate empHireDate;

    // 사용자
    private long userId;
    private String userStatus;
    private LocalDateTime userCreateDate;

    // 권한
    private long roleId;
    private String roleName;
    private char roleStockIn;
    private char roleUpdateMinStock;
    private char roleAddItem;
    private char roleApproveItem;
    private char roleUpdateUserInfo;
}

File: project3/Team3/App_BQ/app/src/main/java/bitc/full502/app_bq/data/model/UserDto.kt
package bitc.full502.app_bq.data.model

data class UserDto(
    val deptId: Long,
    val deptCode: String?,
    val deptName: String?,

    val empId: Long,
    val empCode: String,
    val empName: String,
    val empEmail: String?,
    val empPhone: String?,
    val empBirthDate: String?,   // LocalDate → String("yyyy-MM-dd")
    val empHireDate: String?,    // LocalDate → String

    val userId: Long,
    val userStatus: String?,
    val userCreateDate: String?, // LocalDateTime → String("yyyy-MM-dd'T'HH:mm:ss")

    val roleId: Long,
    val roleName: String?,
    val roleStockIn: Char,
    val roleUpdateMinStock: Char,
    val roleAddItem: Char,
    val roleApproveItem: Char,
    val roleUpdateUserInfo: Char
)
